Assets/Scripts/Bird.cs

```diff
@@ -114,19 +117,34 @@ private void OnCollisionEnter2D(Collision2D collision)
         if(ready)
         {
             ready = false;
-            StartCoroutine(ResetAfterDelay());
+            StartCoroutine(ResetAfterDelay(15));
         }
         
     }
 
-    private IEnumerator ResetAfterDelay()
+    private bool EntitiesAreMoving()
     {
-        yield return new WaitForSeconds(3);
+        Entity[] allEntities = GameObject.FindObjectsOfType<Entity>();
+        foreach (Entity e in allEntities)
+        {
+            if (e.isActiveAndEnabled && !e.isDead && e.GetComponent<Rigidbody2D>().velocity.magnitude > 0.0001)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private IEnumerator ResetAfterDelay(int delay)
+    {
+        yield return new WaitForSeconds(delay);
         rigidbody.isKinematic = true;
         rigidbody.velocity = Vector2.zero;
         rigidbody.position = startPosition;
+        onCooldown = false;
+        ready = true;
 
-        StartCoroutine(ReadyAfterDelay());
+        //StartCoroutine(ReadyAfterDelay());
```

Comment:
This was necessary to avoid a bug with dragging him in between life cycle hooks. If the time can be smaller it is okay to decrease the time.